### PR TITLE
Wr/sf style

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
make `plugin-packaging` friendly for `sf` and backwards compatible for `sfdx`

just pushing so I can see a diff

### What issues does this PR fix or reference?
@W-12083491@

### CODE REVIEW

File: src/commands/force/package/beta/convert.ts [hunk 1]
```diff
@@ -16,63 +21,74 @@ import {
...
+      summary: messages.getMessage('instance'),
+      description: messages.getMessage('longInstance'),
       hidden: true,
     }),
   };
 
   public async run(): Promise<PackageVersionCreateRequestResult> {
+    const { flags } = await this.parse(PackageConvert);
     // eslint-disable-next-line @typescript-eslint/require-await
     Lifecycle.getInstance().on(PackageEvents.convert.progress, async (data: PackageVersionCreateEventData) => {
```
Comment (peternhale) [inline src/commands/force/package/beta/convert.ts:88]: Maybe improve the user experience by replacing the scrolling progress with a spinner?

File: src/commands/force/package/beta/install.ts [hunk 2]
```diff
@@ -88,17 +107,17 @@ export class Install extends SfdxCommand {
 
   public static parseStatus(
     request: PackageInstallRequest,
-    ux: UX,
+    command: Install | Report,
```
Comment (peternhale) [inline src/commands/force/package/beta/install.ts:110]: Can we find a way around passing an instance of the command just to get access to the log method? I see this as the same pattern we using in core; emit a lifecycle event that the command is listening for.

File: src/commands/force/package/beta/installed/list.ts [hunk 3]
```diff
@@ -52,8 +67,10 @@ export class PackageInstalledListCommand extends SfdxCommand {
       SubscriberPackageVersionName: { header: 'Version Name' },
       SubscriberPackageVersionNumber: { header: 'Version' },
     };
-    this.ux.table(records, tableOptions, { 'no-truncate': true });
+    this.table(records, tableOptions, { 'no-truncate': true });
 
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (peternhale) [inline src/commands/force/package/beta/installed/list.ts:73]: @WillieRuemmele why the need for a ts-ignore?

File: src/commands/force/package/beta/version/create.ts [hunk 4]
```diff
@@ -24,184 +28,217 @@ import Package2VersionStatus = PackagingSObjects.Package2VersionStatus;
...
 
-    this.ux.startSpinner(messages.getMessage('requestInProgress'));
+    this.spinner.start(messages.getMessage('requestInProgress'));
+
     const result = await PackageVersion.create(
       {
-        connection: this.hubOrg.getConnection(),
+        connection: flags['target-hub-org'].getConnection(flags['api-version']),
         project: this.project,
-        ...this.flags,
```
Comment (peternhale) [inline src/commands/force/package/beta/version/create.ts:188]:
``` suggestion
...Object.fromEntries(Object.entries(flags).map(([key, value]) => [key.replace(/-/g, ''), value]));
```

File: src/commands/force/package/beta/version/promote.ts [hunk 5]
```diff
@@ -6,51 +6,63 @@
...
       if (packageVersionData.HasMetadataRemoved) {
-        this.ux.warn(messages.getMessage('hasMetadataRemovedWarning'));
+        this.warn(messages.getMessage('hasMetadataRemovedWarning'));
       }
 
       // Prompt for confirmation
-      if (
-        !(await this.ux.confirm(messages.getMessage('packageVersionPromoteConfirm', [this.flags.package as string])))
-      ) {
+      if (!(await this.confirm(messages.getMessage('packageVersionPromoteConfirm', [flags.package])))) {
```
Comment (peternhale) [inline src/commands/force/package/beta/version/promote.ts:63]: I would suggest throwing an error here instead having to ts-ignore the empty return with a message like `User abandoned promote`

**Review by peternhale — CHANGES_REQUESTED** [review]
@WillieRuemmele we have been removing `force` from the file structure to get rid of the `force` topic and then adding a command alias back to the original command.

I have also been trying to add the schema validation during this transition, since it is supported with SfProject.

File: src/commands/package/version/displayancestry.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  SfCommand,
+} from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import { Package, PackageAncestryNodeData } from '@salesforce/packaging';
+
+// Import i18n messages
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_displayancestry');
+
+export type DisplayAncestryCommandResult = PackageAncestryNodeData | string;
```
Comment (peternhale) [inline src/commands/package/version/displayancestry.ts:22]:
``` suggestion
export type DisplayAncestryCommandResult = PackageAncestryNodeData | string | void;
```

File: src/commands/package/version/displayancestry.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        const paths = packageAncestry.getLeafPathToRoot(packageAncestry.requestedPackageId);
+        this.log(`${paths[0].map((p) => p.getVersion()).join(' -> ')} (root)`);
+        this.log();
+      }
+      const treeProducer = packageAncestry.getTreeProducer(!!flags.verbose);
+      if (!flags.json) {
+        treeProducer.produce();
+      }
+    }
+    return jsonProducer.produce() as PackageAncestryNodeData;
```
Comment (peternhale) [inline src/commands/package/version/displayancestry.ts:81]: remove cast

File: src/commands/package/version/list.ts [hunk 8]
```diff
@@ -120,19 +134,19 @@ export class PackageVersionListCommand extends SfdxCommand {
...
-        const aliases = [];
+        const aliases: string[] = [];
         ids.forEach((id) => {
           const matches = project.getAliasesFromPackageId(id);
           if (matches.length > 0) {
-            aliases.push(matches);
+            aliases.concat(matches);
           }
         });
         const AliasStr = aliases.length > 0 ? aliases.join() : '';
```
Comment (peternhale) [inline src/commands/package/version/list.ts:144]: I know this has been this was for a while, but not sure this correct.
The code above finds and retains a list of aliases for a package id.

AliasStr is the join of these aliases with no delimiter, so aliases ['a','b','c'] becomes `abc` which makes no sense.

File: src/commands/package/version/list.ts [hunk 9]
```diff
@@ -120,19 +134,19 @@ export class PackageVersionListCommand extends SfdxCommand {
...
         const ids = [record.Id, record.SubscriberPackageVersionId];
-        const aliases = [];
+        const aliases: string[] = [];
         ids.forEach((id) => {
           const matches = project.getAliasesFromPackageId(id);
           if (matches.length > 0) {
-            aliases.push(matches);
+            aliases.concat(matches);
           }
         });
```
Comment (peternhale) [inline src/commands/package/version/list.ts:143]:
``` suggestion
        const aliases: string[] = ids.map((id) => project.getAliasesFromPackageId(id)).flat();
```

File: src/commands/package/version/promote.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+      const err = SfError.wrap(e as Error);
+      if (err.name === 'DUPLICATE_VALUE' && err.message.includes('previously released')) {
+        err.message = messages.getMessage('previouslyReleasedMessage');
+        err.actions = [messages.getMessage('previouslyReleasedAction')];
+      }
+      throw err;
+    }
+
+    this.log(messages.getMessage('humanSuccess', [result.id]));
+    return result;
```
Comment (peternhale) [inline src/commands/package/version/promote.ts:81]:
``` suggestion
    try {
      const result = await packageVersion.promote();
      result.id = packageVersionData.SubscriberPackageVersionId;
      this.log(messages.getMessage('humanSuccess', [result.id]));
      return result;
    } catch (e) {
      const err = SfError.wrap(e as Error);
      if (err.name === 'DUPLICATE_VALUE' && err.message.includes('previously released')) {
        err.message = messages.getMessage('previouslyReleasedMessage');
        err.actions = [messages.getMessage('previouslyReleasedAction')];
      }
      throw err;
    }
```

File: src/commands/package/version/update.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+import { PackageSaveResult, PackageVersion } from '@salesforce/packaging';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_version_update');
+
+export class PackageVersionUpdateCommand extends SfCommand<PackageSaveResult> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('summary');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly aliases = ['force:package:beta:version:update', 'force:package:version:update'];
```
Comment (peternhale) [inline src/commands/package/version/update.ts:25]: @WillieRuemmele why are the `beta` command aliases being retained to allow customer some time to adjust CI/CD?

File: src/commands/package/version/update.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    });
+    const result = await pv.update({
+      VersionDescription: flags['version-description'],
+      Branch: flags.branch,
+      InstallKey: flags['installation-key'],
+      VersionName: flags['version-name'],
+      Tag: flags.tag,
+    });
+
+    this.log(messages.getMessage('success', [result.id]));
```
Comment (peternhale) [inline src/commands/package/version/update.ts:79]: this.logSuccess

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/commands/package/version/update.ts (hunk 11 above)
Comment (WillieRuemmele) [reply src/commands/package/version/update.ts:25]: they were left as part of the packaging GA

**Review by peternhale — APPROVED** [review]

**Pull request merged**